### ISSUE TITLE
Unify the history of a group before and after it has been upgraded to a supergroup

### ISSUE THREAD

**Issue by mymike00** [tdlib/td#1002]
I've just upgraded a group to a supergroup. Everything fine on telegram desktop (it doesn't even mention the group has been upgraded tbf...).
On [TELEports](https://gitlab.com/ubports/apps/teleports) (which uses tdlib) the history before the upgraded is no longer visible.

I just saw issue #223... Would it be possible for tdlib to do this itself and just expose the history of both chat before and after the upgrade unified?

**Comment by levlam** [comment]
The old messages are still located in the old chat. They are only visually showed to the user as being in the supergroup chat history. They can't be transparently returned in the supergroup's history, because given that they are located in the basic group, they can't be replied or pinned, don't have a public link even the supergroup is public, can be deleted just for self and are different in much more ways from the supergroup messages.

This can be transparently implemented in the future in TDLib, but proper implementation will need really a lot of changes in TDLib's API and internally.

**Comment by mymike00** [comment]
> This can be transparently implemented in the future in TDLib

Yeah, I was asking for this s.t. each client doesn't have to do it itself...

**Comment by Flohack74** [comment]
Humm so the original question would be, how did the old Telegram clients (pre-tdlib) merge these chats transparently? For each message you would need to remember from which chat it came if you want to act upon them. I find this overly complex.

**Comment by levlam** [comment]
@Flohack74 Yes, they do this and much more to merge history of two chats.

**Comment by Ma-colla-coder** [comment]
Quiero mi dinero Javi recarga la cuenta de magico

**Comment by Ma-colla-coder** [comment]
Temprano mañana @md588

**Comment by Ma-colla-coder** [comment]
Cómo hacer un baypas un número que giré como engrane y tendrás super grupos pidiendo reutilizar los heredados

**Comment by Ma-colla-coder** [comment]
Porque necesitas eliminar los grupos heredados, por qué no engancharlos como gusano ,los puedes comprimir.con un botón pusch y marcarlos como arcoiris y extenderlos como abanico ese es tu súper grupo
